src/clj/muuntaja/json.clj

```diff
@@ -99,29 +103,41 @@
   ([] (make-mapper {}))
   ([options]
    (doto (ObjectMapper.)
-     (.registerModule (make-clojure-module options)))))
+     (.registerModule (make-clojure-module options))
+     (cond-> (:pretty options) (.enable SerializationFeature/INDENT_OUTPUT)
+             (:escape-non-ascii options) (.enable ^"[Lcom.fasterxml.jackson.core.JsonGenerator$Feature;"
+                                                  (into-array [JsonGenerator$Feature/ESCAPE_NON_ASCII]))))))
 
 (def ^ObjectMapper +default-mapper+
   "The default ObjectMapper instance used by muuntaja.json/to-json and
   muuntaja.json/from-json unless you pass in a custom one."
-  (make-mapper))
+  (make-mapper {}))
 
 (defn from-json
   "Decode a value from a JSON string or InputStream.
 
   To configure, pass in an ObjectMapper created with make-mapper."
   ([data] (from-json data +default-mapper+))
-  ([data ^ObjectMapper mapper]
-   (if (string? data)
-     (.readValue mapper ^String data ^Class Object)
-     (.readValue mapper ^InputStream data ^Class Object))))
+  ([data arg]
+   (let [decode (fn [data ^ObjectMapper mapper]
```

Comment:
the innerfunction gets generated in all function calls. Could be clearer (and faster) if the mapper is defined in a let used in a body `(let [mapper (cond...)] ....)`